midend/parserUnroll.cpp

```diff
@@ -391,7 +391,7 @@ class ParserSymbolicInterpreter {
             }
 
             if (value == nullptr) value = factory->create(type, true);
-            if (value->is<SymbolicError>()) {
```

Comment:
Well the code below still checks if it is nullptr only only then sets it. That said, if it was nullptr it would crash until now, so presumably it never is :-). It seems to me based on reading the code that the factory could return `nullptr` if this pass was applied to a parser with non-inlined subparser. So that would suggest the pass probably should not be called.

Sadly, this is a complex code that is not well documented. If there was a documented prerequisite that the parsers has to be inlined I'd be happy adding an assert there. But it seems to that at least for the factory, someone was thinking about the case where they are not. I believe this modification is safe, adding the error would be a little bit more risky.

Reply:
Makes sense, the parser unroll pass unfortunately does not have good error checking. I think this error should be handled correctly but this is not the point of this PR. The changes here just make this erroneous behavior explicit.